File: ruby/easy/first_unique_character_in_a_string.rb
# @param {String} s
# @return {Integer}
def first_uniq_char(s)
    map = Hash.new
    s.each_char do |c|
        map[c] = map[c] ? (map[c] + 1) : 1
    end
    s.each_char.with_index do |c,i|
        return i if map[c] == 1
    end
    -1
end

File: javascript/easy/firstUniqueCharacterinaString.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    const map = new Map();
    for(let c of s) {
        const count = map.get(c);
        map.set(c, (count === undefined) ? 1 : count+1);
    }
    for(let i = 0; i < s.length; i++) {
        if(map.get(s[i]) === 1) {
            return i;
        }
    }
    return -1;
};
